app/models/concerns/language_model/export.rb

```diff
@@ -0,0 +1,55 @@
+module LanguageModel::Export
+  extend ActiveSupport::Concern
+
+  def as_json(options = {})
+    options = options.with_indifferent_access
+    attrs = super(options)
+    attrs["api_service_name"] = api_service_name if options[:only].include?(:api_service_name)
+    attrs
+  end
```

Comment:
I got an exception in rails console when I did `LanguageModel.last.as_json` because I left off only. We could just add `&` before, but it technically won't work if we do `...as_json(only: :api_service_name)`

We don't necessarily need to support all those different edge cases, but out of curiosity I was asking chatgpt if there was a more idiomatic way to support delegate attributes with as_json since this is a common pattern. I don't think there is, but ChatGPT wrote this code as a suggestion (I didn't test this):

```
  def as_json(options = {})
    super(options).tap do |hash|
      hash['api_service_name'] = api_service_name if include_api_service_name?(options)
    end
  end

  private

  def include_api_service_name?(options)
    options = options.with_indifferent_access
    return true unless options[:only] || options[:except]

    only = Array(options[:only])
    except = Array(options[:except])

    (only.empty? || only.include?('api_service_name')) && !except.include?('api_service_name')
  end
```

Reply:
Ooh, a little research (surely this is easier than this?) and it's now just:

```ruby
  def as_json(options = {})
    super(options).merge("api_service_name" => api_service_name)
  end
```

Reply:
More research (because, why is this so complicated?) and discovered I can override `attributes` method to add new custom attributes.

```ruby
  def attributes
    super.merge("api_service_name" => api_service_name)
  end
```

However, for reasons I cannot debug, but I've moved on with my life, we also need to re-define this method. Dumb, but whatever.

```ruby
  # Unsure why this needs to re-defined, but the original ActiveModel::Serialization
  # implementation is ignoring the #attributes method above.
  def attribute_names_for_serialization
    attributes.keys
  end
```

Reply:
Very nice, that's really simple.